### PR TITLE
WIP: Set up GitHub Actions to run tests on PRs and Deploy SnapShots on BINTRAY

### PR DESCRIPTION
**Pull request by solemabrothers**

### CODE REVIEW

File: .github/workflows/deploy-snapshots.yml [hunk 1]
```diff
@@ -36,4 +36,4 @@ jobs:
         run: mvn -B deploy --file pom.xml
         env:
           BINTRAY_USERNAME: ugandaemr-ci
```
Comment (ssmusoke) [inline .github/workflows/deploy-snapshots.yml:38]: @solemabrothers Change this username to a Secret too

Comment (ssmusoke) [inline .github/workflows/deploy-snapshots.yml:38]: @solemabrothers Change this username to a Secret too

File: .github/workflows/verify-prs.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+name: Verify PRs
```
Comment (ssmusoke) [inline .github/workflows/verify-prs.yml:1]: @solemabrothers Why verify PRs why not Run Tests?

File: .github/workflows/verify-prs.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+            ${{ runner.os }}-maven-
+        # Set up Java 1.8 with Maven
+      - name: Set up JDK 1.8
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+
+        # Execute the Maven verify command to compile, package, test, verify
+      - name: Build with Maven
+        run: mvn -B verify --file pom.xml
```
Comment (ssmusoke) [inline .github/workflows/verify-prs.yml:31]: @solemabrothers Why not `mvn clean install`

File: .github/workflows/verify-prs.yml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-
+        # Set up Java 1.8 with Maven
+      - name: Set up JDK 1.8
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+
+        # Execute the Maven verify command to compile, package, test, verify
```
Comment (ssmusoke) [inline .github/workflows/verify-prs.yml:29]: @solemabrothers Where is the deployment to BinTray?

Comment (ssmusoke) [inline .github/workflows/verify-prs.yml:29]: @solemabrothers Where is the deployment to BinTray?

File: .github/workflows/deploy-snapshots.yml [hunk 5]
```diff
@@ -36,4 +36,4 @@ jobs:
...
-          BINTRAY_PASSWORD: aGJUeZZbGSy3aGbZc49aEf
+          BINTRAY_PASSWORD: ${{ secrets.BINTRAY_ACCESS_PASSWORD }}
```
Comment (ssmusoke) [inline .github/workflows/deploy-snapshots.yml:39]: @solemabrothers Why use secrets.BINTRAY_ACCESS_PASSWORD here and BINTRAY_PASSWORD elsewhere

Comment (solemabrothers) [reply .github/workflows/deploy-snapshots.yml:39]: @ssmusoke BINTRAY_PASSWORD is the environment variable while secrets.BINTRAY_ACCESS_PASSWORD is the value of the secret created.

Comment (ssmusoke) [reply .github/workflows/deploy-snapshots.yml:39]: @solemabrothers then name the secret secrets.BINTRAY_PASSWORD to match the environment variable

File: .github/workflows/deploy-snapshots.yml (hunk 1 above)
Comment (solemabrothers) [reply .github/workflows/deploy-snapshots.yml]: Done

File: .github/workflows/verify-prs.yml (hunk 2 above)
Comment (solemabrothers) [reply .github/workflows/verify-prs.yml]: Done

File: .github/workflows/verify-prs.yml (hunk 3 above)
Comment (solemabrothers) [reply .github/workflows/verify-prs.yml]: Done

File: .github/workflows/deploy-snapshots.yml (hunk 5 above)
Comment (solemabrothers) [reply .github/workflows/deploy-snapshots.yml]: Done

File: .github/workflows/verify-prs.yml (hunk 4 above)
Comment (solemabrothers) [reply .github/workflows/verify-prs.yml:29]: Explained

Comment (solemabrothers) [reply .github/workflows/verify-prs.yml:29]: Explained

File: .github/workflows/deploy-snapshots.yml (hunk 1 above)
Comment (solemabrothers) [reply .github/workflows/deploy-snapshots.yml]: Done

Comment (solemabrothers) [reply .github/workflows/deploy-snapshots.yml]: Done

File: .github/workflows/deploy-snapshots.yml (hunk 5 above)
Comment (ssmusoke) [reply .github/workflows/deploy-snapshots.yml]: @solemabrothers This is not done - check

**Pull request merged**